source/common/BUILD

```diff
@@ -81,6 +81,30 @@ envoy_cc_library(
     ],
 )
 
+envoy_cc_library(
+    name = "request_source_plugin_impl",
```

Comment:
This is a good point to start thinking about slightly different file / directory structure here. The common directory is growing which makes it harder to navigate. How about we create a new request_source directory instead at the same level where common is?

In addition to this, we should split out the individual request source plugin implementations into their own files. Placing them all into one will quickly outgrow "good file size". There isn't a hard rule, but as a recommendation I usually start thinking about a refactor when a file reaches about 300 lines. Let's try to refactor these as suggested and see how it looks.

Reply:
Understood.